tests/php/Forms/TreeMultiselectFieldTest.php

```diff
@@ -10,22 +10,57 @@ class TreeMultiselectFieldTest extends SapphireTest
 {
     protected static $fixture_file = 'TreeDropdownFieldTest.yml';
 
+    public function testEmptyChoiceReadonly()
+    {
+        $field = new TreeMultiselectField('TestTree', 'Test tree', File::class);
+
+        $asdf = $this->objFromFixture(File::class, 'asdf');
+        $subfolderfile1 = $this->objFromFixture(File::class, 'subfolderfile1');
+
+        $schemaStateDefaults = $field->getSchemaStateDefaults();
+        $this->assertArraySubset(['id' => 'TestTree', 'name' => 'TestTree', 'value' => 'unchanged'], $schemaStateDefaults, $strict = true);
+
+        $items = $field->getItems();
+        $this->assertCount(0, $items, $message = 'there must be no items selected');
+
+        $html = $field->performReadonlyTransformation()->Field();
+
+        $this->assertEquals(
+            <<<"HTML"
+<span id="TestTree_ReadonlyValue" class="readonly">
+	<i>('none')</i>
+</span><input type="hidden" name="TestTree" class="hidden" id="TestTree" />
```

Comment:
Sorry, one last thing - this is weak since it's prone to break if you override the templates that this form field uses to render with. Examples of this are in the CWP recipes, where we modify some of the core form field templates to increase their accessibility compliance. I'd suggest doing partial assertions instead, e.g. `assertContains("('none')", $html)`, `assertNotContains('value="', $html)` etc

Reply:
well, since it's supposed to be a unit test, I'd rather not to test the rendering at all and only test the class own and inherited methods. I'll try to rewrite it in a more generic way, without rendering and templates involved

P.S. even if we test the partials of the resulting string, there are no guarantees it's gonna always contain all the things we are checking for.